query/src/main/scala/filodb/query/exec/rangefn/RangeFunction.scala

```diff
@@ -394,14 +398,35 @@ object LastSampleFunction extends RangeFunction {
   }
 }
 
+object LastSampleInWindowFunction extends RangeFunction {
+  override def needsLastSample: Boolean = true
+  def addedToWindow(row: TransientRow, window: Window): Unit = {}
+  def removedFromWindow(row: TransientRow, window: Window): Unit = {}
+
+  override def apply(startTimestamp: Long,
+            endTimestamp: Long,
+            window: Window,
+            sampleToEmit: TransientRow,
+            queryConfig: QueryConfig): Unit = {
+    if (window.size > 1)
+      throw new IllegalStateException(s"Window had more than 1 sample. Possible out of order samples. Window: $window")
+    // if no sample found in the window, set the value to NaN
+    if (window.size == 0 || startTimestamp > window.head.getLong(0)) {
```

Comment:
`startTimestamp > window.head.getLong(0)` is not needed. Always true.

Reply:
looks like this condition is needed, as we add last sample before start of the window. This is where the logic is to include the last sample outside the window `SlidingWindowIterator.shouldAddCurToWindow`